src/components/data_search/DatasetSearchTableDisplay.tsx

```diff
@@ -48,10 +49,72 @@ interface DatasetSearchTableDisplayProps{
   tab: DatasetSearchTableTab<DatasetTerm | DatasetTerm[]>
 }
 
+interface Sort {
+  colIndex: number;
+  dir: number;
+}
+
+// Helper function to check if a value should be treated as empty/undefined
+const isMissingValue = (value: CellData['value']): boolean => {
+  return isUndefined(value) ||
+    isEmpty(value) ||
+    value === '--'
+};
+
 export const DatasetSearchTableDisplay = (props: DatasetSearchTableDisplayProps) => {
   const { onSelect, exportableDatasets, filteredData, selected, tab } = props;
+  const [sort, setSort] = React.useState<Sort>({ colIndex: -1, dir: 1 });
   const headers = tab.makeHeaders(filteredData, selected, onSelect, exportableDatasets);
-  const rowData = tab.makeRows(filteredData, headers);
+
+
+  const sortData = React.useCallback((data: CellData[][], sortState: Sort) => {
+    if (sortState.colIndex < 0) return data;
+
+    const colIndex = sortState.colIndex;
+    const columnLabel = headers[colIndex]?.label;
+
+    // Check if we're sorting a numeric column like Participants or Participant Count
+    const isNumericColumn = columnLabel === 'Participants' || columnLabel === 'Participant Count';
+
+    return [...data].sort((a, b) => {
+      const aValue = a[colIndex]?.value;
+      const bValue = b[colIndex]?.value;
+
+      // Always sort empty values to the bottom regardless of sort direction
+      // This includes undefined, empty strings, and placeholder values like "--"
+      const aIsEmpty = isMissingValue(aValue);
+      const bIsEmpty = isMissingValue(bValue);
+
+      if (aIsEmpty && bIsEmpty) return 0;
+      if (aIsEmpty) return 1; // Move 'a' to the bottom when it's empty
+      if (bIsEmpty) return -1; // Move 'b' to the bottom when it's empty
+
+      if (isNumericColumn) {
+        const numA = Number(aValue);
+        const numB = Number(bValue);
+
+        // Handle NaN values similar to undefined
+        if (isNaN(numA) && isNaN(numB)) return 0;
+        if (isNaN(numA)) return 1; // Treat NaN like undefined, to the bottom
+        if (isNaN(numB)) return -1; // Treat NaN like undefined, to the bottom
+
+        return (numA - numB) * sortState.dir;
+      }
+      const strA = String(aValue).toLowerCase();
+      const strB = String(bValue).toLowerCase();
```

Comment:
It looks like `aValue` and `bValue` are actually `string` types, so this works, but they could possibly be objects in the future. For example, date values might be added to the table in a future tasks.